zegami_sdk/collection.py

```diff
@@ -1020,6 +1025,35 @@ def __len__(self) -> int:
     def __repr__(self) -> str:
         return "<Collection id={} name={}>".format(self.id, self.name)
 
+    def id_to_class(self, ID):
+        return self.classes[ID - 1]['name']
+
+    def get_annotations_metadata_to_csv(self, outpath='output.csv',
+                                        anno_type=None) -> pd.DataFrame:
+        """Creates a .csv file with annotations metadata from a collection.
+        Defaults to searching for annotations of all types.
+        """
+        metadata = self.get_annotations_metadata(anno_type)
+        metadata.to_csv(outpath)
+        print('Saved annotations metadata at "{}".'.format(outpath))
+        return metadata
+
+    def get_annotations_metadata(self, anno_type=None) -> pd.DataFrame:
```

Comment:
Has this been tested? On mine this does not function - for example `.get_annotations()` result has no `['annotations']` key
Is the metadata column re-ordering tuple specific to one collection?

Reply:
Suggest replacing with
```
def get_annotations_as_dataframe(self, anno_type=None, source=0) -> pd.DataFrame:
    """
    Collects all annotations of a type (or all if anno_type=None) and returns
    the information as a dataframe.
    """
    
    annos = self.get_annotations(anno_type=anno_type)['sources'][source]['annotations']
    to_dict = lambda a: {
        'Row Index': self.imageset_index_to_row_index(a['image_index']),
        'ID': a['id'],
        'Type': a['type'],
        'Author': a['author'],
        'Imageset Index': a['image_index'],
    }
    annos_dictlist = [to_dict(a) for a in annos]
    
    df = pd.DataFrame(annos_dictlist)
    
    return df
```

Reply:
Does this get the annotations['metadata'] as well? It doesn't seem to and it is important for Ebbon... This code was based and tested on annotations manually added on zegami

Reply:
The existing code looks ok, although Doug's suggestion is easier to follow. Would need to include the metadata as Sofia points out.